### PR TITLE
Switch from repository_dispatch to workflow_dispatch

### PR DESCRIPTION
**Pull request by benreynolds-drizly**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,16 +38,15 @@ This action does three things:
...
-    # JSON string to provide as client payload on the repository dispatch event.
-    # Three additional fields are automatically added to the client_payload json object
-    # prior to dispatching:
+    # Inputs to pass to the workflow, must be a JSON encoded string ex. '{ "myinput":"myvalue" }'
+    # Three additional fields are automatically added to the inputs prior to dispatching:
     #    check_id: The ID of the queued GitHub check created by this action
-    #    GITHUB_REPOSITORY: The repository invoking this action, formatted as "<owner>/<repository-name>"
-    #    GITHUB_SHA: The GITHUB_SHA in the workflow invoking this action
-    client_payload: |
+    #    github_repository: The repository invoking this action, formatted as "<owner>/<repository-name>"
```
Comment (ianfixes) [inline README.md:44]: It's mildly weird to do repo/username as separate fields above and then combined here

File: action.yml [hunk 2]
```diff
@@ -20,9 +20,14 @@ inputs:
...
-  event_type:
+  target_ref:
+    required: false
+    description: Ref which should be triggered on the target repository
+    default: main
+
+  workflow_filename:
     required: true
-    description: Name of the repository dispatch event type
+    description: Name of the workflow file to trigger in the target repository (without .yml extension)
```
Comment (ianfixes) [inline action.yml:30]: It sounds like this is saying
> The basename (no .yml extension) of the file in `.github/workflows/` responding to the workflow_dispatch event

File: initialization.go [hunk 3]
```diff
@@ -65,9 +65,14 @@ func getInputs() (inputs, error) {
 		return inputs{}, errors.New("input 'target_owner' not set")
 	}
 
-	eventType, ok := os.LookupEnv("INPUT_EVENT_TYPE")
+	targetRef, ok := os.LookupEnv("INPUT_TARGET_REF")
+	if err != nil {
```
Comment (ianfixes) [inline initialization.go:69]: if `ok != nil`?

File: README.md (hunk 1 above)
Comment (benreynolds-drizly) [reply README.md:44]: I agree it is a bit weird but there are two specific reasons for doing it this way:

1. We only have 10 inputs to work with so we leave more available for actual workflow inputs if we send owner and repo as a single combined input
2. The automatically provided GITHUB_REPOSITORY environment variable is a combination of owner/repo so this aims to emulate that

Would you prefer I combine the actual inputs on this action vs having `target_owner` and `target_repository` separately?

Comment (ianfixes) [reply README.md:44]: I think combining the inputs to this action would probably make sense for consistency's sake, but you raise another point: it sounds like the inputs being passed on to `workflow_dispatch` compete for the 10 inputs to this action... meaning that you would be unable to properly call a 10-input action.  Could we get around that by wrapping the destination payload in a single input and unwrapping it later?

Comment (benreynolds-drizly) [reply README.md:44]: We could kiiiiiinda do that but then the receiving workflow would have a lot of extra responsibility (to parse the inputs) and we wouldn't get the input validation that GitHub provides. In short, [this block of code](https://github.com/DrizlyInc/distributor/blob/br/workflow_dispatch_test/.github/workflows/wait.yml#L5) would not be possible anymore.

**Review by mikekantzer-drizly — APPROVED** [review]
The golang side of this looks good. 

The repo is already renamed.

File: README.md (hunk 1 above)
Comment (ikatz-drizly) [reply README.md:44]: (Page not found for that block of code.)  I guess I was imaging that the _caller_ of the action would stuff the inputs into a JSON block, and the workflow_dispatch action would split it out internally such that the eventual receiver just got it as individual inputs.  The other thing occurring to me is that (as-is) the names of this action's inputs become unusable as inputs to the "awaited" action.  I don't think there's anything for it at this point though.

**Pull request merged**